Lib/lighthouse/lighthouse.cpp

```diff
@@ -217,22 +248,25 @@ void Lighthouse::RunRecordObject() {
   } catch (ImageQualityException e) {
     fprintf(stderr, "Lighthouse::RunRecordObject() encountered an error: %s\n", e.what());
     Feedback::PlaySoundNamed("nothing-recognized");
+    delegate->OnError(LighthouseError::ERROR_COULD_NOT_EXTRACT_DESCRIPTION);
 
-    return; // FIXME: Report actual error.
+    return;
   }
 
   SaveDescription(sourceDescription, source);
 
   Feedback::OnItemRecorded(sourceDescription.GetId());
+  delegate->OnSuccess(source);
```

Comment:
It was my initial idea. However, I feel that some stuff cannot be replaced, e.g. waiting until TTS has finished speaking.